model/riscv_insts_vext_arith.sail

```diff
@@ -420,77 +445,1759 @@ function clause execute(MOVETYPEV(vs1, vd)) = {
+                          /* signed overflow case returns zero naturally as required due to -1 divisor */
+                          to_bits(SEW, r)
+                        },
+        MVV_VREM     => {
+                          let r : int = if signed(vs1_val[i]) == 0 then signed(vs2_val[i]) else rem_round_zero(signed(vs2_val[i]), signed(vs1_val[i]));
+                          /* signed overflow case returns zero naturally as required due to -1 divisor */
+                          to_bits(SEW, r)
+                        }
+      }
+    }
+  };
+
+  write_vreg(num_elem, SEW, LMUL_pow, vd, result);
+  vstart = EXTZ(0b0);
+  RETIRE_SUCCESS
+}
+
+mapping mvvtype_mnemonic : mvvfunct6 <-> string = {
+  MVV_VAADDU   <-> "vaaddu.vv",
+  MVV_VAADD    <-> "vaadd.vv",
+  MVV_VASUBU   <-> "vasubu.vv",
+  MVV_VASUB    <-> "vasub.vv",
+  MVV_VMUL     <-> "vmul.vv",
+  MVV_VMULH    <-> "vmulh.vv",
+  MVV_VMULHU   <-> "vmulhu.vv",
+  MVV_VMULHSU  <-> "vmulhsu.vv",
+  MVV_VDIVU    <-> "vdivu.vv",
+  MVV_VDIV     <-> "vdiv.vv",
+  MVV_VREMU    <-> "vremu.vv",
+  MVV_VREM     <-> "vrem.vv"
+}
+
+mapping clause assembly = MVVTYPE(funct6, vm, vs2, vs1, vd)
+  <-> mvvtype_mnemonic(funct6) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ sep() ^ vreg_name(vs1) ^ maybe_vmask(vm)
+
+/* ************************ OPMVV (MVVtype Multiply-Add) ************************* */
+/* Multiply-Add instructions switch the order of source operands in assembly (vs1/rs1 before vs2) */
+union clause ast = MVVMATYPE : (mvvmafunct6, bits(1), regidx, regidx, regidx)
+
+mapping encdec_mvvmafunct6 : mvvmafunct6 <-> bits(6) = {
+  MVV_VMACC       <-> 0b101101,
+  MVV_VNMSAC      <-> 0b101111,
+  MVV_VMADD       <-> 0b101001,
+  MVV_VNMSUB      <-> 0b101011
+}
+
+mapping clause encdec = MVVMATYPE(funct6, vm, vs2, vs1, vd) if haveRVV()
+  <-> encdec_mvvmafunct6(funct6) @ vm @ vs2 @ vs1 @ 0b010 @ vd @ 0b1010111 if haveRVV()
+
+function clause execute(MVVMATYPE(funct6, vm, vs2, vs1, vd)) = {
+  let SEW      = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let VLEN     = int_power(2, get_vlen_pow());
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+
+  let 'n = num_elem;
+  let 'm = SEW;
+
+  let vm_val  : vector('n, dec, bool)     = read_vmask(num_elem, vm, vreg_name("v0"));
+  let vs1_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs1);
+  let vs2_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs2);
+  let vd_val  : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vd);
+  result      : vector('n, dec, bits('m)) = undefined;
+  mask        : vector('n, dec, bool)     = undefined;
+
+  (result, mask) = init_masked_result(num_elem, SEW, LMUL_pow, vd_val, vm_val);
+
+  assert(VLEN >= 0);
+  foreach (i from 0 to (num_elem - 1)){
+    if mask[i] then{
+      result[i] = match funct6 {
+        MVV_VMACC  => slice(to_bits(VLEN, signed(vs1_val[i]) * signed(vs2_val[i])), 0, SEW) + vd_val[i],
```

Comment:
This `slice(to_bits(...), ...)` pattern appears several times.  I think you could do it more simply by using `get_slice_int` directly (or perhaps by sticking to bitvectors throughout rather than round-tripping through integers).  In this particular example it would help Sail's translation to some theorem provers because they would currently need to case split on all the possible values for `VLEN`.

Reply:
This pattern is modified to use `get_slice_int` now.